src/StructuredLogViewer/Controls/BuildControl.xaml.cs

```diff
@@ -1024,11 +1030,30 @@ private object FindInFiles(string searchText, int maxResults, CancellationToken
                     return null;
                 }
 
-                var haystack = file.Value;
-                var resultsInFile = haystack.Find(searchText);
-                if (resultsInFile.Count > 0)
+                if (!string.IsNullOrEmpty(searchText) && searchText.StartsWith("$secret"))

```

Comment:
this condition will run for every file, and it doesn't change. How about we extract a bool above the foreach and just check it here?